File: DailyPractices/Leetcode_3814/brute.java
class Solution {
    public int maxCapacity(int[] costs, int[] capacity, int budget) {
        int n = costs.length;
        int maxCapacity = 0;

        // Pick one item
        for (int i = 0; i < n; i++) {
            if (costs[i] < budget) {
                maxCapacity = Math.max(maxCapacity, capacity[i]);
            }
        }

        // Pick two items
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) { // j = i+1 to avoid same item
                int totalCost = costs[i] + costs[j];
                int totalCapacity = capacity[i] + capacity[j];
                if (totalCost < budget) {
                    maxCapacity = Math.max(maxCapacity, totalCapacity);
                }
            }
        }

        return maxCapacity;
    }
}

File: LeetCode/DailyPractices/Leetcode_3814/brute.cpp
class Solution {
public:
    int maxCapacity(vector<int>& costs, vector<int>& capacity, int budget) {
        int n = costs.size();
        int maxCap = 0;

        // pick 1 item
        for (int i = 0; i < n; i++) {
            if (costs[i] < budget) maxCap = max(maxCap, capacity[i]);
        }

        // pick 2 items
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int totalCost = costs[i] + costs[j];
                if (totalCost < budget) {
                    maxCap = max(maxCap, capacity[i] + capacity[j]);
                }
            }
        }

        return maxCap;
    }
};